com.microsoft.java.test.runner/src/main/java/com/microsoft/java/test/runner/TestOutputStream.java

```diff
@@ -0,0 +1,107 @@
+package com.microsoft.java.test.runner;
+
+import java.io.PrintStream;
+import java.util.StringJoiner;
+
+public class TestOutputStream implements TestStream {
+    private final PrintStream out;
+    private final PrintStream err;
+    private static final TestOutputStream instance = new TestOutputStream();
+
+    private TestOutputStream() {
+        this.out = System.out;
+        this.err = System.err;
+    }
+
+    public static TestOutputStream instance() {
+        return instance;
+    }
+
+    @Override
+    public void print(TestMessageItem item) {
+        String content = toJson(item);
+        if (item.type == TestMessageType.Error) {
+            this.err.print(content);
+        } else {
+            this.out.print(content);
+        }
+    }
+
+    @Override
+    public void println(TestMessageItem item) {
+        String content = toJson(item);
+        if (item.type == TestMessageType.Error) {
+            this.err.println(content);
+        } else {
+            this.out.println(content);
+        }
+    }
+
+    @Override
+    public void flush() {
+        this.out.flush();
+        this.err.flush();
+    }
+
+    private static String toJson(TestMessageItem item) {
+        StringBuilder builder = new StringBuilder("@@<TestRunner-{\"name\":");
+        builder.append('"').append(item.name).append('"');

```

Comment:
What is the issue for using GSON?

Reply:
GSON is an external dependency, I have to pack the dependency into the jar file and it would conflict for some projects. I'm still investigating the root cause, but I'd like to change it to previous implementation before the root cause is clear.